content/Proposals/Done/202106-proposals-process.md

```diff
@@ -0,0 +1,112 @@
+
+### Pitfalls of the current solution
+
+Currently, the Observability Platform team have the process defined [here (internal)](https://docs.google.com/document/d/1eojXStPdq1hYwv36pjE-vKR1q3dlBbpIx5w_L_v2gNo/edit#heading=h.kpdg1wrd3pcc), whereas the In-Cluster part were not defining any official process ([as per here (internal)](https://docs.google.com/document/d/1vbDGcjMjJMTIWcua5Keajla9FzexjLKmVk7zoUc0_MI/edit#heading=h.n0ac5lllvh13)).
+
+In practice, both teams had somehow similar flow:
+
+* For upstream: Follow the upstream project's contributing guide, e.g Thanos
+* For downstream:
+  * Depending on the size:
+    * Small features can be proposed during the bi-weekly team-sync or directly in Slack.
+      * If the team can reach consensus in this time, then document the decision somewhere written, e.g. an email, Slack message to which everyone can add an emoji reaction, etc.
+      * Add a JIRA ticket to plan this work.
+    * Large features might need a design doc:
+      1. Add a JIRA ticket for creating the design doc
+      2. Create a new Google Doc in the team folder based on [this template](https://docs.google.com/document/d/1ddl_dLxjoIvWQuRgLdzL2Gd1EX1mkJQUZ-rgUh-T4d8/edit)
+      3. Fill sections
+      4. Announce it on the team mailing list and Slack channel
+      5. Address comments / concerns 6 Define what "done" means for this proposal, i.e. what is the purpose of this design document:
+      * Knowledge sharing / Brain dump: This kind of document may not need a thorough review or any official approval
+      * Long term vision and Execution & Implementation: If approved (with LGTM comments, or in an approved section) by a majority of the team and no major concerns consider it approved. NOTE: The same applies to rejected proposals.
+      1. If the document has no more offline comments and no consensus was reached, schedule a meeting with interested parties.
+      2. When the document changes status, move it to the appropriate status folder in the design docs directory of the team folder. If an approved proposal concerns a component with its own directory, e.g. Telemeter, then create a shortcut to the proposal document in the component-specific directory. This helps us find design documents by topic and by status.
+
+It served us well, but it had the following issues (really similar to ones stated in [handbook proposal](../Accepted/202106-handbook.md#pitfalls-of-the-current-solution)):
+
+* Even if our Google Design docs organized in our team drive, those Google documents are not easily discoverable.
+* Existing Google doc-based documents are hard to consume. The formatting is widely different. Naming is inconsistent.
+* Document creation is rarely actionable. There is no review process, so the effort of creating a relevant document might be wasted, as the document is lost. This also leads to docs being in the half-completed state, demotivating readers to look at it.
```

Comment:
How will the new process ensure that people drive their proposals to completion?

Reply:
Very solid point. It's always based on authors motivation. Usually, the problem is not author motivation but lack of process, lack of visibility or decision making time. I think this proposal fixes those things.

It does not fix the motivation part. If author does not want to finish the proposal, maybe it's not that important and it will get just stale PR, closed after some time of inactivitiy

Reply:
In my opinion, the guarantee of a process that says "your doc will be reviewed by a human in X time and you will be given feedback" is a good motivation. This, coupled with "your work is recognized and you can add it to your list of achievements" is much more than we currently have and should motivate the completion of designs more than we already do. The goals is not to ensure 100% completion of a design doc; it's impossible and not part of anyone's job description. But a structures process should create a positive difference IMO

Reply:
Added summary